client/components/data/query-users-suggestions/index.jsx

```diff
@@ -0,0 +1,57 @@
+/**
+ * External dependencies
+ */
+import { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { isRequestingUserSuggestions as isRequesting } from 'state/users/suggestions/selectors';
+import { requestUserSuggestions } from 'state/users/suggestions/actions';
+
+class QueryUsersSuggestions extends Component {
+	componentWillMount() {
+		this.request( this.props );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( this.props.siteId === nextProps.siteId ) {
+			return;
+		}
+
+		this.request( nextProps );
+	}
+
+	request( props ) {
+		if ( props.isRequesting ) {
+			return;
+		}
+
+		props.requestUserSuggestions( props.siteId );
+	}
+
+	render() {
+		return null;
+	}
+}
+
+QueryUsersSuggestions.propTypes = {
+	siteId: PropTypes.number,
+	isRequesting: PropTypes.bool,
+	requestUserSuggestions: PropTypes.func
+};
+
+QueryUsersSuggestions.defaultProps = {
+	requestUserSuggestions: () => {},
+	isRequesting: false
+};
```

Comment:
Optional but we can also inline these using class properties. These are still stage-2, but we do use a [babel transform](https://babeljs.io/docs/plugins/transform-class-properties/) for this

So this might look like:
```
class QueryUsersSuggestions extends Component {
    static propTypes = {
         siteId: PropTypes.number,
         isRequesting: PropTypes.bool,
         requestUserSuggestions: PropTypes.func
    };
    
    static defaultProps = {
        requestUserSuggestions: () => {},
        isRequesting: false
    };
//...
```